src/plugins/MamicoCoupling.h

```diff
@@ -2,114 +2,135 @@
-	 * 
-	 * Following steps are taken, if coupling is switched on:
-	 * - Distribute incoming momentum among affected cells
-	 * - Apply boundary force to molecules near microscopic domain boundary
-	 * */	
-	void afterForces(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			unsigned long simstep
-	) override;
-
-	void endStep(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			Domain* domain, unsigned long simstep) override;
-
-	void finish(ParticleContainer* particleContainer,
-			DomainDecompBase* domainDecomp, Domain* domain) override;
-
-	std::string getPluginName() override {
-		return std::string("MamicoCoupling");
-	}
-
-	static PluginBase* createInstance() { return new MamicoCoupling(); }
+  MamicoCoupling() {}
+  virtual ~MamicoCoupling() {}
+
+  /**
+   * Prints to log that mamico coupling is initialized
+   */
+  void init(ParticleContainer *particleContainer,
+            DomainDecompBase *domainDecomp, Domain *domain) override;
+
+  void readXML(XMLfileUnits &xmlconfig) override;
+
+  void beforeEventNewTimestep(ParticleContainer *particleContainer,
+                              DomainDecompBase *domainDecomp,
+                              unsigned long simstep) override;
+
+  /**
+   * Takes coupling steps such as particle insertion, to make sure they are
+   * accounted for before forces are calculated.
+   *
+   * Following steps are taken, if coupling is switched on:
+   * - Iterate over cells to average values like momentum and mass, to pass to
+   * macroscopic solvers
+   * - Distribute incoming mass from macroscopic solver by inserting perticles
+   * (if enabled)
+   * - Run the MaMiCo thermostat cell by cell
+   *
+   * The distributeMass method calls the
+   * updateParticleContainerAndDecomposition() function at the end, so we end up
+   * with a container with halo particles present. Hence a manual halo clearance
+   * is done to restore the state of the container.
+   */
+  void beforeForces(ParticleContainer *particleContainer,
+                    DomainDecompBase *domainDecomp,
+                    unsigned long simstep) override;
+
+  /**
+   * Performs adjustments after force calculation
+   *
+   * Following steps are taken, if coupling is switched on:
+   * - Distribute incoming momentum among affected cells
+   * - Apply boundary force to molecules near microscopic domain boundary
+   */
+  void afterForces(ParticleContainer *particleContainer,
+                   DomainDecompBase *domainDecomp,
+                   unsigned long simstep) override;
+
+  void endStep(ParticleContainer *particleContainer,
+               DomainDecompBase *domainDecomp, Domain *domain,
+               unsigned long simstep) override;
+
+  void finish(ParticleContainer *particleContainer,
+              DomainDecompBase *domainDecomp, Domain *domain) override;
+
+  std::string getPluginName() override { return std::string("MamicoCoupling"); }
```

Comment:
```suggestion
  std::string getPluginName() override { return "MamicoCoupling"; }
```
No need to repeat the return type.